### DISCUSSION TITLE
Apache 开源社群的“石头汤” | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#28] [Comments]
# Apache 开源社群的“石头汤” | 夜天之书

《程序员修炼之道》讲了一个有趣的“石头汤”寓言。这个寓言里，饿着肚子的外来人在村子里烧了一锅水，放了三块石头，开始煮“石头汤”。这样的行为引来好奇的村民围观，外来人顺势在“石头汤”的基础上引导村民们添加食材以改善这锅料理。最后，村民和外来人一起煮出了一锅靓汤，外来人于是把石头从汤里扔掉，所有人分享了这顿美餐。 开源协同的工作方式与制作“石头汤”的方式有些相似。开源社群的核心成员与寓言中的外来人一样

https://www.tisonkun.org/2022/06/15/stone-soup-apache/

**Comment by Xuanwo** [comment]
一个很吊诡的现状是有些机构各种委员会成立了一大堆，召开一个会议说项目开源了，然后就开始希望社区自己运转起来产出代码

**Reply by ZoomQuiet** [reply]
#是也乎, (￣▽￣)

然也, 这才是正版的 "石头汤"

这里只是比喻, FLOSS 中的 石头汤, 这石头本身可就是有功能的, 可以正确丢出来打猎的...

> ...在一个基本可用的软件的基础上，抛出可以做的更好的可能性，身体力行并团结潜在的开发者一起不断实现做出的预言

这种版本的 "石头汤" 已经能从石头中种出无限菜品了, 无论外界是否给东西,
类似 OpenSSL, cURL, logj... 都是这种石头, 根本不是普通纯粹委员会能作到的事儿...

**Comment by Xuanwo** [comment]
分享一个有趣的案例

![image](https://user-images.githubusercontent.com/5351546/190425550-efee48b5-fbf8-4d77-a639-771dab29523f.png)

https://github.com/Xuanwo/serde-env 是一个从环境变量中解析数据的 serde 实现，我在做好了基础的功能之后，社区自己为了满足自己的需求主动进行了贡献
